presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1033,20 +1051,43 @@ protected Object visitNode(Node node, Object context)
             return ImmutableList.copyOf(Iterables.transform(ImmutableList.copyOf(types), Functions.forMap(expressionTypes)));
         }
 
-        private boolean hasUnresolvedValue(Object... values)
+        private Object invokeOperator(OperatorType operatorType, List<? extends Type> argumentTypes, List<Object> argumentValues)
+        {
+            Signature operatorSignature = metadata.getFunctionRegistry().resolveOperator(operatorType, argumentTypes);
+            return invoke(session, metadata.getFunctionRegistry().getScalarFunctionImplementation(operatorSignature), argumentValues);
+        }
+
+        private boolean hasUnresolvableValue(Object... values)
         {
-            return hasUnresolvedValue(ImmutableList.copyOf(values));
+            List<Object> valuesList = Arrays.stream(values)
+                    .filter(value -> value != null)
+                    .collect(Collectors.toList());
+
+            return !valuesList.isEmpty() && hasUnresolvableValue(valuesList);
         }
 
-        private boolean hasUnresolvedValue(List<Object> values)
+        private boolean hasUnresolvableValue(List<Object> values)
         {
-            return any(values, instanceOf(Expression.class));
+            return values.stream()
+                    .filter(value -> !(value instanceof Literal))
+                    .anyMatch(instanceOf(Expression.class)::apply);
         }
 
-        private Object invokeOperator(OperatorType operatorType, List<? extends Type> argumentTypes, List<Object> argumentValues)
+        private Object resolveLiteral(Object value)
```

Comment:
We need this because the pattern in ExpressionInterpreter is now that anybody that expects to get a Java object back from a call to `process` can now get back a literal instead. And we cannot invoke functions on literals.